### PR TITLE
ConvNets lecture

### PR DESCRIPTION
**Pull request by matteo-d-m**
Hey Luigi, 

This is to propose my lecture on convolutional neural networks. 

As you will see, there's quite a lot of material. The plan is to do one hour theory and one hour code. No practicals, but suggested exercises at the end of the lecture.   

I could not compromise more on the theory - if I did, things would have not made sense. I can only promise I'll be clear and to the point! 

Let me know what you think,
Matteo

### CODE REVIEW

File: lectures/Lecture3.3_ConvNets.ipynb [hunk 1]
```diff
@@ -0,0 +1,2382 @@
...
+    "\n",
+    "print(f\"Neuron's activation value (before any nonlinearity): {neuronal_activation} | After sigmoid: {with_sigmoid_activation}\") \n",
+    "print(f\"Neuron's activation value (before any nonlinearity): {neuronal_activation} | After ReLU: {with_relu_activation}\") "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Interim Summary: The Artificial Neuron in a Nutshell**\n",
```
Comment (vigji) [inline lectures/Lecture3.3_ConvNets.ipynb:340]: Maybe this clarifying picture could be move above for people who can't grasp quickly the math

File: lectures/Lecture3.3_ConvNets.ipynb [hunk 2]
```diff
@@ -0,0 +1,2382 @@
...
+    "where \n",
+    "\n",
+    "- $ a_i = $ the activation value of the $ i^{th} $ neuron"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here's what all looks like in video:\n",
```
Comment (vigji) [inline lectures/Lecture3.3_ConvNets.ipynb]: Do you have a link for the video? It won't play on my machine (I could see the video from the file but maybe good for other readers to link it as well)

File: lectures/Lecture3.3_ConvNets.ipynb [hunk 3]
```diff
@@ -0,0 +1,2382 @@
...
+    "This operation is known as _**(discrete) convolution**_ and is equivalent to taking the dot product between a patch of input and a set of weights."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that:\n",
+    "\n",
+    "- Convolution returns weighted sums of inputs that are specific to a given input region, that is, that represent a specific _feature_\n",
```
Comment (vigji) [inline lectures/Lecture3.3_ConvNets.ipynb:729]: The video is good but I feel that a video where one see (graphically) how matching kernel and underlying values result in high values for that entry of the convolution would make the explanation stronger for who isn't familiar with matrices

File: lectures/Lecture3.3_ConvNets.ipynb [hunk 4]
```diff
@@ -0,0 +1,2382 @@
...
+    "                                          transform=transforms.ToTensor())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The variables that we created are _dataset objects_. A dataset object implements two crucially important methods:\n",
+    "\n",
+    "- `__len__()`, which allows users to get the dataset's length\n",
```
Comment (vigji) [inline lectures/Lecture3.3_ConvNets.ipynb]: We did not cover those special methods in the objects class, so I would keep their explicit mentioning out as it's distracting - sufficient to say you can index them and retrieve the Len with `len()`

**Review by vigji — COMMENTED** [review]
I left some small comments here and there. Structure is very good, and people will surely appreciate the very explicit slides! Very well done!!

Only one thing: squeezing the content in a single lecture is ambitious, and I'm afraid it will be challenging to pack everything - I'm wondering if in the practical you could get to a first run of the network before talking about model selection, which is important but could be left at the end in case you don't get there!

File: lectures/Lecture3.3_ConvNets.ipynb (hunk 1 above)
Comment (matteo-d-m) [reply lectures/Lecture3.3_ConvNets.ipynb:340]: Mh, I understand your concern but I would leave it there. I think it's easier/better to go from math to picture than vice versa & I want to emphasize that the artificial neuron is just a simple mathematical operation. I'll make sure I am understood!

File: lectures/Lecture3.3_ConvNets.ipynb (hunk 2 above)
Comment (matteo-d-m) [reply lectures/Lecture3.3_ConvNets.ipynb]: I am trying with links to the GitHub folder

File: lectures/Lecture3.3_ConvNets.ipynb (hunk 3 above)
Comment (matteo-d-m) [reply lectures/Lecture3.3_ConvNets.ipynb:729]: I am sorry, I don't understand what you mean :/

File: lectures/Lecture3.3_ConvNets.ipynb (hunk 4 above)
Comment (matteo-d-m) [reply lectures/Lecture3.3_ConvNets.ipynb]: Ok

**Comment by matteo-d-m** [comment]
Thanks for the comments, I am applying most of them. The new version will be ready in a bit. Shall I open a new pull request when ready?

**Pull request closed without merging**